dynamallow/types/base.py

```diff
@@ -0,0 +1,29 @@
+class BaseModel(object):
+    """``BaseModel`` is the base class for ``types.Model``. It must define ``dynamallow_validate`` which runs validation
+    in your desired serialization library, ``dynamallow_fields`` which returns a dictionary of key value pairs
+    where keys are attributes and values are the type of the attribute, and ``field_to_dynamo_type`` which returns
+    the dynamo type character for the input type (we implement schematics_ and marshmallow_).
+
+    .. _schematics: https://schematics.readthedocs.io/en/latest/
+    .. _marshmallow: https://marshmallow.readthedocs.io/en/latest/
+    """
+    @staticmethod
+    def field_to_dynamo_type(field):
+        """ Returns the dynamo type character given the field. """
+        raise NotImplementedError('Child class must implement field_to_dynamo_type')
+
+    @classmethod
+    def dynamallow_fields(cls):
+        """ Returns a dictionary of key value pairs where keys are attributes and values are type classes """
+        raise NotImplementedError('{} class must implement dynamallow_fields'.format(cls.__name__))
+
+    @classmethod
+    def dynamallow_validate(cls, obj):
```

Comment:
In Python classes, variables and functions are treated the same way - as attributes on the class. So if `marshmallow.Schema` or `schematics.Model` put an attribute `fields` or `validate` on their class (which they do), our function would shadow the parent method or variable.